### PR TITLE
DDF-2861 Refactor RouteManager to use Felix ArtifactInstaller

### PR DESCRIPTION
**Pull request by harrison-tarr**
@jrnorth
@dcruver

### CODE REVIEW

File: broker/broker-routemanager/pom.xml [hunk 1]
```diff
@@ -52,6 +52,11 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-log4j12</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.fileinstall</artifactId>
+            <version>3.5.4</version>
```
Comment (ryeats) [inline broker/broker-routemanager/pom.xml]: should this match the karaf version?

**Review by dcruver — APPROVED** [review]

File: distribution/ddf-common/src/main/resources/etc/org.apache.felix.fileinstall-routes.cfg [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# *
+# * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+# * <http://www.gnu.org/licenses/lgpl.html>.
+# *
+# */
+
+felix.fileinstall.dir = ${karaf.base}/etc/routes
+felix.fileinstall.tmpdir  = ${karaf.data}/generated-bundles
+felix.fileinstall.poll    = 997
```
Comment (jrnorth) [inline distribution/ddf-common/src/main/resources/etc/org.apache.felix.fileinstall-routes.cfg:16]: Might want to add a filter, e.g. `felix.fileinstall.filter = .*\\.xml`

Comment (harrison-tarr) [reply distribution/ddf-common/src/main/resources/etc/org.apache.felix.fileinstall-routes.cfg:16]: Doesn't the `canHandle` call accomplish that?

File: broker/broker-routemanager/src/main/java/org/codice/ddf/broker/routemanager/DynamicRouteDeployer.java [hunk 3]
```diff
@@ -13,134 +13,102 @@
...
-    public void destroy() throws Exception {
-        camelContext.stop();
+    @Override
+    public boolean canHandle(File file) {
+        return file.getName()
+                .endsWith(".xml");
     }
+
+    public Map<File, List<RouteDefinition>> getProcessedFiles() {
+        if (processedFiles == null) {
```
Comment (jrnorth) [inline broker/broker-routemanager/src/main/java/org/codice/ddf/broker/routemanager/DynamicRouteDeployer.java]: Why not do `private Map<File, List<RouteDefinition>> processedFiles = new HashMap<>();` instead so you can use the field directly without having to use another method that checks for nulls?

Comment (harrison-tarr) [reply broker/broker-routemanager/src/main/java/org/codice/ddf/broker/routemanager/DynamicRouteDeployer.java]: I was going to override that method in my unit test for the `update` method but then I ended up just using mocks. It seemed a little cleaner to use the method, but It certainly is simpler to just create a `new HashMap<>()`. Fixed.

File: distribution/ddf-common/src/main/resources/etc/org.apache.felix.fileinstall-routes.cfg (hunk 2 above)
Comment (harrison-tarr) [reply distribution/ddf-common/src/main/resources/etc/org.apache.felix.fileinstall-routes.cfg:16]: Fixed.

**Review by ryeats — APPROVED** [review]

**Pull request merged**